### DISCUSSION TITLE
posts/dbos

### DISCUSSION THREAD

**Discussion by giscus** [whexy/whexy-blog-comments#14] [General]
# posts/dbos

通过在 VoltDB 上构建一个 PoC，研究者们证明了可以用分布式数据库提供诸如IPC、任务调度、文件系统的服务。这些服务甚至获得了媲美单机服务的性能。使用数据表实现的 MQ/IPC 服务，某些场景下性能超过了 gRPC。基于数据表的文件系统写入速度超过了 EXT4。

https://www.whexy.com/posts/dbos

**Comment by MstMoonshine** [comment]
1. 架构图中虽然 Microkernel 画在最底层，但是看起来逻辑上真正的老大是 level 2 的 DBMS？
2. 页表管理在传统操作系统中是个核心点，但是在 DBOS 里，它真的还那么重要吗？对 DBOS 来说，页表管理是不是可以归类为硬件相关的功能，跟 IO 管理一起下放到 Microkernel 里，而不去操心它的细节？如果进程管理提高到集群这个 level，我们是不是甚至可能抛弃页表，直接跑嵌入式 RTOS 作为 Microkernel？
3. 我们的确不能编译器确定运行时所需内存大小。但是考虑现有的集群架构，比如K8S，每个 container 创建之初应该也是有一个内存上限的，所以 either 现有架构里存在同样的问题 or 这不是个重要的问题？注意到作者给出了一个重要的前提：*每个子任务占用一段较短的时间*，在这个前提下，使用一个较高的 upper bound 占有一块较大的内存是否也不会带来很大的代价？

**Reply by MstMoonshine** [reply]
关于第三点的补充，也许我们应该用 Memory Size * Memory Occupation Time 作为新指标来衡量内存消耗，同时考虑空间和时间。（实际情况这个公式应该是个积分而不是乘法。）
